src/Math-Matrix/PMTensor.class.st

```diff
@@ -0,0 +1,145 @@
+Class {
+	#name : #PMTensor,
+	#superclass : #Object,
+	#instVars : [
+		'array',
+		'shape',
+		'first',
+		'strides'
+	],
+	#category : #'Math-Matrix'
+}
+
+{ #category : #'instance creation' }
+PMTensor class >> newWith: anInteger [
+
+	^ self new array: {anInteger} withShape: #(  )
+]
+
+{ #category : #'instance creation' }
+PMTensor class >> rows: anArray [
```

Comment:
It would help me understand this message better if the argument name was clearer

Reply:
This is due to the fact that we choose to browse the tensors according to the rows

Reply:
So what does `anArray` mean? Are you really referring to a row? If that is true, a rename would support my understanding

Reply:
I agree this name is misleading. Also,for ranks > 2, there are no rows or columns any  more.

Suggestion: fromNestedArray